src/color.rs

```diff
@@ -731,6 +731,14 @@ impl<T: Primitive> Blend for Rgba<T> {
     fn blend(&mut self, other: &Rgba<T>) {
         // http://stackoverflow.com/questions/7438263/alpha-compositing-algorithm-blend-modes#answer-11163848
 
+        if other.0[3].is_zero() {
+            return;
+        }
+        if other.0[3].to_i32().unwrap() == 255 {
```

Comment:
I believe this comparison should be `other.0[3]` against `T::DEFAULT_MAX_VALUE` as this isn't only being used for `T=u8`.

Reply:
This should work because the impl requires `T: Primitive` which includes `T: PartialOrd+PartialEq`. We might also use a greater-equal comparison instead, but I'm not certain how that actually will work with floating point. Maybe check if the math for floats even works right now?

Reply:
Thanks for the suggestion, I forgot DEFAULT_MAX_VALUE exists 😄